app/frontend/src/features/messages/groupchats/GroupChatView.tsx

```diff
@@ -18,30 +18,43 @@ import PageTitle from "../../../components/PageTitle";
 import { useHistory, useParams } from "react-router-dom";
 import { Skeleton } from "@material-ui/lab";
 import HeaderButton from "../../../components/HeaderButton";
+import Menu, { MenuItem } from "../../../components/Menu";
+import InviteDialog from "./InviteDialog";
+import LeaveDialog from "./LeaveDialog";
+import MembersDialog from "./MembersDialog";
+import GroupChatSettingsDialog from "./GroupChatSettingsDialog";
 
 const useStyles = makeStyles((theme) => ({
   header: { display: "flex", alignItems: "center" },
   title: {
     flexGrow: 1,
     marginInline: theme.spacing(2),
   },
-  menuItem: {
-    paddingInline: theme.spacing(2),
-  },
 }));
 
 export default function GroupChatView() {
   const classes = useStyles();
 
   const menuAnchor = useRef<HTMLAnchorElement>(null);
-  const [menuOpen, setMenuOpen] = useState(false);
+  const [isOpen, setIsOpen] = useState({
+    menu: false,
+    invite: false,
+    members: false,
+    settings: false,
+    leave: false,
+  });
 
-  const handleClick = () => {
-    setMenuOpen(true);
+  const handleClick = (what: keyof typeof isOpen) => {
```

Comment:
```suggestion
  const handleClick = (item: keyof typeof isOpen) => {
```
Slightly clearer since the callback is receiving back which item on the menu it's getting back?